File: ChatGPT/ChatGPT/java/3-wayHandShake.java
import java.io.*;
import java.net.*;

public class ThreeWayHandshakeClient {
    public static void main(String[] args) {
        final String serverAddress = "127.0.0.1";
        final int serverPort = 12345;

        try {
            // 소켓 생성
            Socket socket = new Socket(serverAddress, serverPort);

            // 데이터 입력 및 출력 스트림 생성
            PrintWriter out = new PrintWriter(socket.getOutputStream(), true);
            BufferedReader in = new BufferedReader(new InputStreamReader(socket.getInputStream()));

            // SYN 패킷 전송
            out.println("SYN");
            System.out.println("SYN sent");

            // SYN-ACK 패킷 수신
            String response = in.readLine();
            System.out.println("SYN-ACK received: " + response);

            // ACK 패킷 전송
            out.println("ACK");
            System.out.println("ACK sent");

            // 연결 완료 메시지 수신
            response = in.readLine();
            System.out.println("Connection established: " + response);

            // 소켓 및 입출력 스트림 닫기
            socket.close();
            out.close();
            in.close();
        } catch (IOException e) {
            e.printStackTrace();
        }
    }
}

File: ChatGPT/ChatGPT/php/3-wayHandShake.php
<!DOCTYPE html>
<html>
<head>
    <title>WebSocket Example</title>
    <script type="text/javascript">
        // 웹 소켓 생성
        var socket = new WebSocket("ws://127.0.0.1:12345");

        // 연결이 열렸을 때 호출되는 이벤트 핸들러
        socket.onopen = function(event) {
            console.log("Connection opened");

            // SYN 패킷 전송
            socket.send("SYN");
            console.log("SYN sent");
        };

        // 메시지를 수신했을 때 호출되는 이벤트 핸들러
        socket.onmessage = function(event) {
            console.log("Message received: " + event.data);

            if (event.data === "SYN-ACK") {
                // ACK 패킷 전송
                socket.send("ACK");
                console.log("ACK sent");
            } else if (event.data === "Connection established") {
                // 연결 완료
                console.log("Connection established");

                // 소켓 연결 종료
                socket.close();
            }
        };

        // 연결이 닫혔을 때 호출되는 이벤트 핸들러
        socket.onclose = function(event) {
            console.log("Connection closed");
        };
    </script>
</head>
<body>
    <h1>WebSocket Example</h1>
</body>
</html>
